server/models/loaders/AtLoader.js

```diff
@@ -35,6 +41,8 @@ const AtLoader = () => {
             return ats;
         }
     };
+
+    return singletonInstance;
```

Comment:
As far I can tell, the caching is no different than [the current implementation](https://github.com/w3c/aria-at-app/blob/56205a32f904964dbb36c943d4e3fdb9d2522ce9/server/models/loaders/AtLoader.js#L3). The only difference is that I am using a singleton whereas `main` passes the single instance around using context. But yes, it is possible that this could get out of sync if new Ats were added (though no UI for this exists to my knowledge). Regardless, the singleton doesn't make changes relevant to this so if changes are needed they might be out of scope for this PR.

Reply:
Ah ok I see! There actually is UI for editing the AT versions, it's on the test queue page under the Manage Assistive Technology Versions section. Therefore I do think we're at risk of losing functionality here. Thoughts?

Reply:
Thanks for this!

You are right that this caching is too aggressive if it is possible to modify these values from the client. The current system in `main` uses a similar caching strategy but relies on side effects from a rerender that create a new context object and reset the cache. This feels fragile and will break if there is ever a tool that updates ats or browsers without rerendering the whole tree.

I added a commit ([f53015a](https://github.com/w3c/aria-at-app/pull/879/commits/f53015abc8bd0f85fabe76306b8ffc6c708bbaa7)) that does a manual cache clearing whenever there is an update to the browsers or ats. This isn't the prettiest addition but I do think surfacing some intent with the cache is useful. It also handles regressions with client-side at version modification. Let me know what you think.